ext/drreg/drreg.c

```diff
@@ -65,17 +65,65 @@
 
 #define PRE instrlist_meta_preinsert
 
-/* This should be pretty hard to exceed as there aren't this many GPRs */
-#define MAX_SPILLS (SPILL_SLOT_MAX + 8)
+/* drreg uses DR spill slots by default if no addressable TLS
```

Comment:
Style: case

Instead, can we say sth like "This is a hardcoded upper limit of how many slots drreg is able to track. The tool is responsible to reserve enough slots as described in the docs about ops.num_spill_slots".

All the other questions are answered in other parts of the doc. At the description of ops.num_spill_slots for example.

WDYT?

Reply:
Yes, I agree. However, I had a read over the drreg [documentation](http://dynamorio.org/docs/struct__drreg__options__t.html) again, and ops.num_spill_slots does not explicitly state that the tool *has* to reserve enough slots - only for cross instructions spilling.

> The number of TLS spill slots to use. This many slots will be requested from DR via dr_raw_tls_calloc(). Any slots needed beyond this number will use DR's base slots, which are not allowed to be used across application instructions. DR's slots are also more expensive to access (beyond the first few). This number should be computed as one plus the number of simultaneously used general-purpose register spill slots, as drreg reserves one of the requested slots for arithmetic flag preservation.
> 
> For each simultaneous value that will be held in a register across application instructions, an additional slot must be requested for adjusting the saved application value with respect to application reads and writes.